### PR TITLE
[Reviewer: Matt] Remove the connection pool - just rely on standard SIP DNS resolution

### PR DESCRIPTION
**Pull request by rkd-msw**
We discussed the other day that the connection pool probably wasn't as useful as when it was originally implemented, because our DNS resolution function allows us to spread load across a cluster rather than routing all traffic to one peer.

I've therefore tried just cutting it out. I've verified that this does mean Bono retries requests when a Sprout is down, fixing #1050 and (maybe) #1495 - see SAS screenshot below.

2016-07-29_18h06_28

As discussed, this probably has slightly worse behaviour when under so much load that you have TCP congestion (as we only have one connection per Sprout, not 50 connections total), but i think that's the right tradeoff - given that Bono is more for demos/getting started than production, it's better to be resilient (and capable of demoing Clearwater's fault-tolerance) than optimising for behaviour under TCP congestion.

### CODE REVIEW

File: src/bono.cpp [hunk 1]
```diff
@@ -3268,24 +3252,6 @@ pj_status_t init_stateful_proxy(SubscriberDataManager* reg_sdm,
...
   dialog_tracker = new DialogTracker(flow_table);
 
-  // Create a connection pool to the upstream proxy.
-  if (upstream_proxy_connections > 0)
-  {
-    pjsip_host_port pool_target;
-    pool_target.host = pj_strdup3(stack_data.pool, upstream_proxy_arg.c_str());
-    pool_target.port = upstream_proxy_port;
-    sprout_ip_tbl = SNMP::IPCountTable::create("bono_connected_sprouts",
-                                               ".1.2.826.0.1.1578918.9.2.3.1");
```
Comment (mirw) [inline src/bono.cpp]: What are we doing about this statistic?  Do we need to maintain it?  (Probably not.)  If not, though, we should tidy up the MIB definition, and any Cacti scripts that use it (as well as anything else).

**Comment by mirw** [comment]
Looks OK - one comment about statistics inline.  However, I'd also like to confirm that this isn't going to break our stress runs - how do we do that?

**Comment by rkd-msw** [comment]
I'll test that we can drive Sprout nodes to 60% CPU both with and without this change - is there any kind of breakage you can think of which that won't catch?

**Comment by mirw** [comment]
I guess overload is probably the interesting one - do we recover OK from TCP congestion?  (Maybe this isn't critical given that it's Bono.)

**Comment by rkd-msw** [comment]
Given that this is Bono-only, I can't convince myself it's worth fixing up the merge conflicts or doing a lot of testing here - closing.

**Pull request closed without merging**